src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -12,6 +12,55 @@ namespace System.Text.Json.Tests
+            yield return new object[] { "\"1997-07-16T19:20\"", "1997-07-16T19:20" };
+            yield return new object[] { "\"1997-07-16T19:20:30\"", "1997-07-16T19:20:30" };
+            yield return new object[] { "\"1997-07-16T19:20:30.45\"", "1997-07-16T19:20:30.45" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555\"", "1997-07-16T19:20:30.4555555" };
+
+            // Skip test T24:00 till #35830 is fixed.
+            // yield return new object[] { "\"1997-07-16T24:00\"", "1997-07-16T24:00" };
+
+            // Test fraction rounding.
+            yield return new object[] { "\"1997-07-16T19:20:30.45555554\"", "1997-07-16T19:20:30.45555554" };
+            // We expect the parser to truncate. `DateTime(Offset).Parse` will round up to 7dp in this case,
+            // so we pass a string representing the Datetime(Offset) we expect to the `Parse` method.
+            yield return new object[] { "\"1997-07-16T19:20:30.45555555\"", "1997-07-16T19:20:30.4555555" };
+
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+01:00\"", "1997-07-16T19:20:30.4555555+01:00" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-01:00\"", "1997-07-16T19:20:30.4555555-01:00" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+04:30\"", "1997-07-16T19:20:30.4555555+04:30" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-04:30\"", "1997-07-16T19:20:30.4555555-04:30" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+0100\"", "1997-07-16T19:20:30.4555555+01:00" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-0100\"", "1997-07-16T19:20:30.4555555-01:00" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555+0430\"", "1997-07-16T19:20:30.4555555+04:30" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555-0430\"", "1997-07-16T19:20:30.4555555-04:30" };
+        }
+
+        public static IEnumerable<object[]> InvalidISO8601Tests()
+        {
+            // Test junk data appended.
+            yield return new object[] { "\"0997-07-160997-07-16\"" };
+            yield return new object[] { "\"0997-07-16abc\"" };
+            yield return new object[] { "\"0997-07-16,0997-07-16\"" };
+            yield return new object[] { "\"1997-07-16T19:20abc\"" };
+            yield return new object[] { "\"1997-07-16T19:20, 123\"" };
+
+            // Other invalid strings.
+            yield return new object[] { "\"997-07-16\"" };
+            yield return new object[] { "\"1997-07\"" };
+            yield return new object[] { "\"1997-7-06\"" };
+            yield return new object[] { "\"1997-07-16T\"" };
+            yield return new object[] { "\"1997-07-6\"" };
+            yield return new object[] { "\"1997-07-6T01\"" };
+            yield return new object[] { "\"1997-07-16Z\"" };
+            yield return new object[] { "\"1997-07-16+01:00\"" };
```

Comment:
Additional invalid tests: space instead of T
`1997-07-16T19:20:30.4555555+`
`1997-07-16T19:20:30.4555555+01:`

etc.